agent-control/src/sub_agent/on_host/command/command_os.rs

```diff
@@ -75,14 +76,47 @@ impl CommandOSNotStarted {
 ////////////////////////////////////////////////////////////////////////////////////
 
 impl CommandOSStarted {
-    pub(crate) fn wait(mut self) -> Result<ExitStatus, CommandError> {
-        self.process.wait().map_err(CommandError::from)
-    }
-
     pub fn get_pid(&self) -> u32 {
         self.process.id()
     }
 
+    pub fn shutdown(&mut self) -> Result<(), CommandError> {
+        let pid = self.get_pid() as i32;
+
+        use nix::{sys::signal, unistd::Pid};
+        signal::kill(Pid::from_raw(pid), signal::SIGTERM)
+            .map_err(|err| CommandError::NixError(err.to_string()))?;
```

Comment:
Do you know if `self.process.kill()` would work in the same way? I was wondering if using it would easy the future Windows support.

Reply:
We can discuss it tomorrow. It's true that we are being very aggressive, but to be honest, I don't know how much of this is relevant in real use.

Reply:
After researching for a bit and talking with an AI, I decided to put the linux implementation we had back. Now we have both a linux and windows implementation. https://github.com/newrelic/newrelic-agent-control/pull/1782/commits/9c3d85311708c93410550e6b7c1ffb399779dc9a.

I added a TODO to properly design the unix/windows shutdown once we implement windows. I expect this to happen shortly. In https://new-relic.atlassian.net/jira/software/c/projects/NR/boards/61?quickFilter=7808&selectedIssue=NR-476370.